products/template/build_scripts/transform.py

```diff
@@ -37,30 +37,27 @@ def _run_dbt_step(run_args=list[str]):
 
 
 def staging():
-    _run_step("staging")
+    _invoke_dbt(["test", "--select", "source:*"])
+    _execute_sql_script("staging")
 
 
 def intermediate():
-    _run_step("libraries")
-    _run_step("green_spaces")
-    _run_step("historic_landmarks")
+    _execute_sql_script("libraries")
+    _execute_sql_script("green_spaces")
+    _execute_sql_script("historic_landmarks")
+    _invoke_dbt(["test", "--select", "libraries green_spaces historic_landmarks"])
 
 
 def product():
-    _run_step("templatedb")
-    _run_step("aggregation/templatedb_boroughs")
-
-
-def test():
-    _run_dbt_step(["deps"])
-    _run_dbt_step(["debug"])
-    _run_dbt_step(["test", "--select", "source:*"])
-    _run_dbt_step(["test", "--select", "templatedb"])
-    _run_dbt_step(["test", "--select", "aggregation"])
+    _execute_sql_script("product/templatedb")
+    _execute_sql_script("product/aggregation/templatedb_boroughs")
+    _invoke_dbt(["test", "--select", "product.templatedb"])
+    _invoke_dbt(["test", "--select", "product.aggregation"])
 
 
 if __name__ == "__main__":
+    _invoke_dbt(["deps"])
```

Comment:
How does line 59 work? Trying to understand package management between our env and the `template
/packages.yml` file

Reply:
it parses a project's `packages.yml` file and installs the listed packages in the same directory as that project. so since I've run this locally, I have a folder that's `template/dbt_packages/dbt_utils/` so the project can use `dbt_utils`

this may be more ideal to do during build environment setup, but it has to be specific to the dbt project (the folder `products/template`)